chat-widget/src/components/webchatcontainerstateful/webchatcontroller/middlewares/renderingmiddlewares/mesageTimestampMiddleware.tsx

```diff
@@ -0,0 +1,108 @@
+import { IWebChatAction } from "../../../interfaces/IWebChatAction";
+import { WebChatActionType } from "../../enums/WebChatActionType";
+import { Constants } from "../../../../../common/Constants";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unused-vars
+const createMessageTimeStampMiddleware = ({ dispatch }: { dispatch: any }) => (next: any) => (action: IWebChatAction) => {
+    if (isApplicable(action)) {
+        return next(evaluateTagsAndOverrideTimeStamp(action));
+    }
+    return next(action);
+};
+
+const isApplicable = (action: IWebChatAction): boolean => {
+    return action.type === WebChatActionType.DIRECT_LINE_INCOMING_ACTIVITY &&
+        isPVAConversation(action) &&
+        isPayloadValid(action) &&
+        isValidChannel(action);
+};
+
+const isPayloadValid = (action: IWebChatAction) => {
+    return action &&
+        action.payload &&
+        action.payload.activity;
+};
+
+const isValidChannel = (action: IWebChatAction): boolean => {
+    return action &&
+        action.payload &&
+        action.payload.activity &&
+        action.payload.activity.channelId &&
+        action.payload.activity.channelId === Constants.acsChannel;
```

Comment:
This project supports "?". action?.payload?.activity?.channelId === Constants.acsChannel. Can you please simply all such expressions in this file?

Reply:
done